src/Bridge/Doctrine/ODM/Types/AbstractEnumType.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+/*
+ * This file is part of the "elao/enum" package.
+ *
+ * Copyright (C) Elao
+ *
+ * @author Elao <contact@elao.com>
+ */
+
+namespace Elao\Enum\Bridge\Doctrine\ODM\Types;
+
+use Doctrine\ODM\MongoDB\Types\ClosureToPHP;
+use Doctrine\ODM\MongoDB\Types\Type;
+use Elao\Enum\EnumInterface;
+
+/**
+ * @template TEnum of EnumInterface
+ */
+abstract class AbstractEnumType extends Type
+{
+    use ClosureToPHP;
+
+    /**
+     * What should be returned on null value from the database.
+     *
+     * @return mixed
+     */
+    protected function onNullFromDatabase()
+    {
+        return null;
+    }
+
+    /**
+     * What should be returned on null value from PHP.
+     *
+     * @return mixed
+     */
+    protected function onNullFromPhp()
+    {
+        return null;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function convertToDatabaseValue($value)
+    {
+        if (null === $value) {
+            return $this->onNullFromPhp();
+        }
+
+        $class = static::getEnumClass();
+
+        if ($value instanceof $class) {
+            return $value->getValue();
+        }
```

Comment:
This check is necessary because you may pass both enum and its value: `$qb->field('foo')->equals(Gender::MALE)` or `$qb->field('foo')->equals(Gender::MALE())`.